src/v/cluster/topic_table_probe.cc

```diff
@@ -9,14 +9,72 @@
 
 #include "cluster/topic_table_probe.h"
 
+#include "cluster/logger.h"
 #include "cluster/topic_table.h"
 #include "config/configuration.h"
+#include "config/node_config.h"
 #include "prometheus/prometheus_sanitize.h"
 
 namespace cluster {
 
 topic_table_probe::topic_table_probe(const topic_table& topic_table)
-  : _topic_table(topic_table) {}
+  : _topic_table(topic_table) {
+    setup_metrics();
+}
+
+void topic_table_probe::setup_metrics() {
+    if (
+      config::shard_local_cfg().disable_public_metrics()
+      || ss::this_shard_id() != 0) {
+        return;
+    }
+    namespace sm = ss::metrics;
+    _metrics.add_group(
```

Comment:
Seastar adds a `shard` label for every metric. I don't think we care about it here, so we should aggregate it away.

```
sm::make_gauge(...).aggregate({sm::shard_label});

Reply:
done
But also there is check in the begining ss::this_shard_id() != 0. So only shard 0 will publish these metrics

Reply:
That's true. In this case aggregation just drops the label, but it's still a good thing to do.